### ISSUE TITLE
ISLCModel.getMaxForce() is wrong

### ISSUE THREAD

**Issue by zepumph** [phetsims/inverse-square-law-common#76]
Discovered because of constants added by @jbphet in `ForceSoundGenerator` as part of https://github.com/phetsims/gravity-force-lab-basics/issues/166. 

the max force is actually calculated from the min distance when the radius is constant.

**Comment by zepumph** [comment]
The bug was that ISLC.getMaxForce assumed that the largest mass and the closest they could get was the max force, but actually when constant radius is on, the densest balls can actually get much closer, by a force factor of ~4. The above logic fix changed the range of a few mapped view elements from about 0-1000 to 0-4000. The effected items were:

* The force arrows above each mass. They are now smaller by default, and change less. To address this large change, I increased the min arrow size so that it starts a bit bigger. @arouinfar will you please confirm that this is acceptable. 
* The puller description bullet in the scene summary changed because the largest region, `Robots pull very hard keeping spheres in place`, only occurs when constant radius is checked. To "hollywood" this, I changed the range from ~0-2000 so that it isn't quite as hard to get the higher values. @terracoda please confirm that this is an alright change.
* @jbphet I changed the use of constants in `ForceSoundGenerator` to use the model. I verified that they are the same values, but feel free to spot check the refactor if you would like.

I think this blocks publication of GFLB because of its potential impact on two views of the sim.

**Comment by zepumph** [comment]
To test, please compare phettest:
http://phettest.colorado.edu/gravity-force-lab-basics/gravity-force-lab-basics_en.html?ea&brand=phet
http://phettest.colorado.edu/gravity-force-lab-basics/gravity-force-lab-basics_a11y_view.html?brand=phet&ea

 to dev 41:
https://phet-dev.colorado.edu/html/gravity-force-lab-basics/1.0.0-dev.41/phet/gravity-force-lab-basics_a11y_view.html
https://phet-dev.colorado.edu/html/gravity-force-lab-basics/1.0.0-dev.41/phet/gravity-force-lab-basics_all_phet.html

**Comment by terracoda** [comment]
@zepumph, there is now a strong visual force change when spheres are "closest" with Constant Size checked and the user unchecks Constant Size:
<img width="685" alt="Screen Shot 2019-07-24 at 2 26 53 PM" src="https://user-images.githubusercontent.com/1902116/61793329-30f39600-ae1f-11e9-9d71-b04e778dcd76.png">
<img width="688" alt="Screen Shot 2019-07-24 at 2 27 19 PM" src="https://user-images.githubusercontent.com/1902116/61793331-3224c300-ae1f-11e9-9f29-d18d290781a1.png">

**Comment by terracoda** [comment]
Ooops, I wasn't ready to post. Will add comments...

**Comment by terracoda** [comment]
@zepumph, I started a new issue so this issue can focus on your original questions.

**Comment by terracoda** [comment]
@zepumph, so I was already a little concerned with a visual mismatch between the described size of the arrows and the actual on-screen size. I think, this change accentuates the visual mismatch. We might have to adjust the ranges for all description scales that are connected to the size of the force.

For example, after the Constant Size checkbox is changed to not checked, the force and the force arrows are much smaller than they were before that change. I would no longer describe the force arrows as "huge" in the second screenshot.

**Comment by terracoda** [comment]
@zepumph, what if we adjusted the lower end of the scale as well.
Visually, I would only like to describe the arrows as "huge" if they are almost off the screen or off the screen. But in terms of newtons, I don't actually know what "huge" is.

**Comment by terracoda** [comment]
@zepumph, when I originally created the force arrow size descriptions and the counter force of the robots pullin in GFL Regular, I thought these descriptions would be directly related, somewhat like this:


Size of Force VectorsFor comparison | Estimated range in newtons | Effort of Robots(counter force)
-- | -- | --
tiny | 0.000 289 - 0.041 712 | pull very little
very small | 0.041 713 - 0.074 154 | pull a little
small | 0.074 155 - 0.260 697 | pull lightly
medium size | 0.260 698 - 0.789 804 | pull gently
large | 0.789 805 - 1.564 181 | pull moderately
very large | 1.564 182 - 2.309 287 | pull hard
huge | 2.309 288 - 11.586 528 | pull very hard

The ranges for the force are a little arbitrary and guess-timated. If I understand the role of teh robots correctly, would we need the same counter force as force to keep the spheres in place?

**Comment by terracoda** [comment]
Given, a "huge force vector", I thought the robots would have to "pull very hard" to keep spheres in place? In the second screenshot above, this pairing is not correct, if I am understanding this correctly?

**Comment by jessegreenberg** [comment]
Great catch @zepumph, thanks. A lot of discussion happened [here](https://github.com/phetsims/coulombs-law/issues/27) and [here](https://github.com/phetsims/inverse-square-law-common/issues/21) about arrow scale design, we probably want to make sure that those decisions are preserved. I think the most important thing was "When the arrows are on screen, the arrow size should scale with force." So if the force doubles, the arrow size should visibly double as well.

**Comment by arouinfar** [comment]
@zepumph I'm on mobile, so I can't check phettest, but I will try to review tonight.

@terracoda the ranges in https://github.com/phetsims/inverse-square-law-common/issues/76#issuecomment-515022872 should be evenly divided. As @jessegreenberg mentioned, the arrow scaling is linear.

> If I understand the role of teh robots correctly, would we need the same counter force as force to keep the spheres in place?

That's correct. Each robot position should map to a single size of force, so from that perspective, your table looks good.

**Comment by zepumph** [comment]
>"When the arrows are on screen, the arrow size should scale with force." So if the force doubles, the arrow size should visibly double as well.

The scale is still linear, the range has just quadrupled, so what used to be the max (~1000) is now about one third of the screen width. 

I'm not really sure how to proceed with this. To me it feels like we should get the visuals how we want it, and then worry about matching the description regions back onto the visuals more accurately. With that in mind I think this is waiting for a review by @arouinfar.

**Comment by arouinfar** [comment]
@zepumph the current max length is too short. Compare latest (top) to master (bottom). When the force is at its max, it's fine for the arrow to spill out-of-bounds. I would double the current max length to give us more range.

![image](https://user-images.githubusercontent.com/8419308/62077639-d20b9380-b207-11e9-9184-2b40531e6e96.png)

@zepumph it also seems like the figure's stance is not being properly mapped. In the screenshot above, the figure should be in its deepest stance. The behavior in latest is correct, but not master.

**Comment by zepumph** [comment]
@arouinfar and I discussed this on zoom. We thought through the best ways forward, and think that it makes sense to have a bit of "hollywooding" in the range maximum for the puller and arrows. Before this bug fix, we were treating all values greater than the max force possible with constant size not checked as the largest region (i.e. longest arrow and pulliest puller). So now we are going to do something similar by passing a fraction of the max force, perhaps .5 or .25, to these visual items to get the linear relationships to look good within the inverse square relationship of the force. 

@arouinfar here is the line of code to change: https://github.com/phetsims/inverse-square-law-common/blob/673c5fc75fc854afdaff973be099f40dc1b94333/js/view/ISLCObjectNode.js#L141

We also discussed the min arrow width I added above, and noted that for CL we definitely need that to be 0, so I reverted it. We will continue that discussion over in https://github.com/phetsims/inverse-square-law-common/issues/77. 

Over to @arouinfar to determine the empirical coefficient to change the max range by.

**Comment by arouinfar** [comment]
@zepumph I think the approach of dividing `modelMaxForce` was a bit naive. I tried something like this:

```js
const modelMaxForce = model.getMaxForce() / 2; 
```

That didn't seem to make any difference, so I tried dividing by 1000 instead, which still didn't do anything. Perhaps there's something we're missing?

**Comment by terracoda** [comment]
@arouinfar, I think in one of your comments you said both distance and mass had linear relationships, but early in my design work, jesse or @zepumph made these plots for me. Based on these plots, I was always under the impression that distance seemed to have a more dramatic effect on the force vectors. I think the first plot is for distance and the second one is for mass?
<img width="367" alt="Screen Shot 2019-08-03 at 9 56 27 AM" src="https://user-images.githubusercontent.com/1902116/62409284-ffc05600-b5d4-11e9-824c-6cf1ae7d2f45.png">

**Comment by terracoda** [comment]
Of course, the plot should now go to 4000 instead of 1000 if I am understanding the error correctly.

**Comment by arouinfar** [comment]
> I think in one of your comments you said both distance and mass had linear relationships,

Sorry for the confusion @terracoda. What I meant is that the arrow size should scale linearly with the **force**. The force itself scales linearly with mass or inversely with radius (as in the plots you shared). In previous versions of GFL (predating ISLC siblings) the force vector did not scale linearly with the force, and the conversations around changing this were linked in @jessegreenberg's https://github.com/phetsims/inverse-square-law-common/issues/76#issuecomment-515118508.

What @zepumph and I discussed leading up to https://github.com/phetsims/inverse-square-law-common/issues/76#issuecomment-516171227 was to introduce a ceiling on the vector size. With the larger `modelMaxForce` we lose some of the dynamic range in the vector size across most interactions. The goal is to include a ceiling for the maximum vector size, at which point increasing the force will not increase the vector size. We would choose an appropriately large ceiling so that the vector is off-screen, so 99.9999% of visual users would never realize that we stopped increasing the vector size. This means that the ranges in https://github.com/phetsims/inverse-square-law-common/issues/76#issuecomment-515022872 would have a larger bin for huge to match.

Before reaching the ceiling, the force vector would scale linearly with the force value. Maintaining the linearity is important. If a student doubles the mass, they would see the force value and vector both double.

**Comment by zepumph** [comment]
>That didn't seem to make any difference, so I tried dividing by 1000 instead, which still didn't do anything. Perhaps there's something we're missing?

@arouinfar I fear this may be a caching problem. I was just able to do this locally on my side just fine. One you change that value, try adding a random query parameter to the url, like `?fjdskalfds` and see if that fixes it. You can also try refreshing multiple times in a row, or go to a new `incognito/private` window. See how that goes.

**Comment by zepumph** [comment]
Above I committed the change to clamp the force by half. Perhaps that can help testing on phettest. Let me know how else I can help.

**Comment by zepumph** [comment]
Currently this is one of the only issues GFLB RC is waiting on.

**Comment by zepumph** [comment]
@arouinfar is still out of town this week. It would be nice to move on with GFLB RC before she gets back. We had discussed clamping together in a meeting, so I feel good about moving forward with this solution. The only tweak to be made is the clamping coefficient applied. This number would be easy enough to maintenance release if it needs to be changed. So I will leave this open and assigned to @arouinfar to tweak if needed, but for now I will move forward.

**Comment by zepumph** [comment]
Furthermore I feel good about this approach because it is the same as what is currently published in GFL regular.

**Comment by zepumph** [comment]
Today At design meeting, @arouinfar @kathy-phet and @ariel-phet dove into this, and we came up with a new strategy for writing this linear function. Commiting the first draft now

Furthermore, @ariel-phet mentioned that perhaps the best way to map the arrow length is with two linear functions. He and @kathy-phet are going to work on that more.

**Comment by zepumph** [comment]
@kathy-phet please use phettest to determine the next steps for these arrow tweaks.

**Comment by zepumph** [comment]
@kathy-phet and I had a meeting about this today where we found values. Commits coming shortly.

**Comment by zepumph** [comment]
There are now two mapping functions set up in GFL(B). @arouinfar and @terracoda please review phettest. CL should be the exact same as before these commits.

**Comment by zepumph** [comment]
@arouinfar do the visual arrows look good in GFL(B) now? Can this issue be closed?

**Comment by zepumph** [comment]
Oh I just found this review from @ariel-phet:

>I looked at both GFL and GFL:B on master
>
>I think the scaling is generally good, and I double checked that it is linear. I think this is good enough for a dev delivery.
>
>@kathy-phet @arouinfar If I were polishing, at "default" I think it would be good for a bit of a tail on the vector arrow. In GLF:B this could be accomplished simply by making the default masses 3 and 6 (billion). That keeps fairly similar goals to using 2 and 4
>
>Again, in GFL, perhaps start with something like 100 kg and 400 kg as the default, that would be a nicer beginning arrow size without have to continue messing with the scaling.

What do you think @arouinfar?

**Comment by arouinfar** [comment]
@zepumph the arrow scaling in GFL(B) is generally looking pretty good. I do find the size of the vectors in the default configuration to be a bit small, however.

For GFL, I think we can go ahead and bump up the default masses to 100 kg and 400 kg and that would take care of the scaling issues.

For GFLB, the range of masses is much smaller, so I don't really want to increase the defaults. Is there any way to make the vector sizes at the default values look a bit closer to how it was in an earlier dev, such as [dev.24](https://phet-dev.colorado.edu/html/gravity-force-lab-basics/1.0.0-dev.24/phet/gravity-force-lab-basics_en_phet.html)?

master
![image](https://user-images.githubusercontent.com/8419308/64731628-d7166080-d49e-11e9-8f74-1e1191874a76.png)

dev.24
![image](https://user-images.githubusercontent.com/8419308/64731655-df6e9b80-d49e-11e9-8377-cece08c42bfc.png)

**Comment by arouinfar** [comment]
GFL and GFLB are looking great @zepumph! Since the changes have all been committed to master, I'll go ahead and close.
